include/knowhere/dataset.h

```diff
@@ -275,6 +292,15 @@ GenResultDataSet(const int64_t rows, const int64_t dim, const void* tensor) {
     return ret_ds;
 }
 
+inline DataSetPtr
+GenResultDataSet(const int64_t rows, const void* centroid_id_mapping) {
+    auto ret_ds = std::make_shared<DataSet>();
+    ret_ds->SetRows(rows);
+    ret_ds->SetCentroidIdMapping(centroid_id_mapping);
+    ret_ds->SetIsOwner(true);
+    return ret_ds;
+}
+
```

Comment:
Let's put this Genxxx function to the user side.

Reply:
we could reuse current dataset actually